src/ir_Sanyo.h

```diff
@@ -34,64 +34,79 @@
 #include "IRsend_test.h"
 #endif
 
+/// Native representation of a Sanyo A/C message.
+union SanyoProtocol{
+  uint8_t raw[kSanyoAcStateLength];  ///< The state in IR code form.
+  // Ref: https://docs.google.com/spreadsheets/d/1dYfLsnYvpjV-SgO8pdinpfuBIpSzm8Q1R5SabrLeskw/edit?usp=sharing
+  struct {
+    // Byte 0
+    uint8_t :8;  // 0x6A (Fixed?)
+    // Byte 1
+    uint8_t Temp :5;
+    uint8_t      :3;
+    // Byte 2
+    uint8_t SensorTemp :5;
+    uint8_t            :1;
+    uint8_t Beep       :1;
+    uint8_t            :1;
+    // Byte 3
+    uint8_t OffHour :4;
+    uint8_t         :4;
+    // Byte 4
+    uint8_t Fan      :2;
+    uint8_t OffTimer :1;
+    uint8_t          :1;
+    uint8_t Mode     :3;
+    uint8_t          :1;
+    // Byte 5
+    uint8_t SwingV :3;
+    uint8_t        :3;
+    uint8_t Power  :2;
+    // Byte 6
+    uint8_t       :3;
+    uint8_t Sleep :1;
+    uint8_t       :4;
+    // Byte 7
+    uint8_t :8;
+    // Byte 8
+    uint8_t Sum :8;
+  };
+  struct {
+    uint8_t :8;
+    uint8_t :8;
+    // Byte 2
+    uint8_t        :2;
+    uint8_t Sensor :1;
+  };
+};
+
 // Constants
 
-// Sanyo A/C
-// Ref: https://docs.google.com/spreadsheets/d/1dYfLsnYvpjV-SgO8pdinpfuBIpSzm8Q1R5SabrLeskw/edit?usp=sharing
-// Byte[0] - 0x6A (Fixed?)
-// Byte[1] - Address + Temperature
-const uint8_t kSanyoAcTempByte = 1;    ///< Index
-const uint8_t kSanyoAcTempOffset = 0;  ///< Mask 0b000xxxxx
-const uint8_t kSanyoAcTempSize = 5;    ///< Mask 0b000xxxxx
 const uint8_t kSanyoAcTempMin = 16;    ///< Celsius
 const uint8_t kSanyoAcTempMax = 30;    ///< Celsius
 const uint8_t kSanyoAcTempDelta = 4;   ///< Celsius to Native Temp difference.
-// Byte[2] - Ambient Temp + Sensor
-const uint8_t kSanyoAcSensorByte = 2;       ///< Index
-const uint8_t kSanyoAcSensorBit = 2;  ///< Mask 0b00x00000
```

Comment:
is this a bug?

Reply:
Indeed it is. Well spotted!
Comment is correct. constant is wrong.
Should be `5`
See https://docs.google.com/spreadsheets/d/1dYfLsnYvpjV-SgO8pdinpfuBIpSzm8Q1R5SabrLeskw/edit#gid=1050142776&range=AI84:AI85